packages/aws-cdk-lib/aws-ecs/README.md

```diff
@@ -623,6 +623,26 @@ taskDefinition.addContainer('windowsservercore', {
 });
 ```
 
+### Using Windows authentication with gMSA
+
+Amazon ECS supports Active Directory authentication for Linux containers through a special kind of service account called a group Managed Service Account (gMSA). For more details, please see the [product documentation on how to implement on Windows containers](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/windows-gmsa.html), or this [blog post on how to implement on  Linux containers](https://aws.amazon.com/blogs/containers/using-windows-authentication-with-gmsa-on-linux-containers-on-amazon-ecs/).
+
+```ts
+declare const taskExecutionRole: iam.Role;
+declare const taskDefinition: ecs.TaskDefinition;
+
+// A task execution role is needed, and it should have permissions to read from the S3 bucket or SSM parameter where the CredSpec file is stored.
+taskDefinition.executionRole = taskExecutionRole
+
+taskDefinition.addContainer('gmsa-container', {
+  image: ecs.ContainerImage.fromRegistry("amazon/amazon-ecs-sample"),
+  cpu: 128,
+  memoryLimitMiB: 256,
+  credentialSpecs: ['credentialspecdomainless:arn:aws:s3:::bucket_name/key_name'], 
+  // Valid values are: 'credentialspecdomainless:ARN' or 'credentialspec:ARN'
```

Comment:
An idea here. What if we exposed two properties, `credentailSpecs` and `credentialSpecsDomainless`, both which just takes the `ARN` component (to be clearly documented in the docstring). And then we can append the correct prefix before sending it to ecs. This will save users from having to do the appending themselves, right?

Reply:
I think this is exactly what we want and this exact care to concoct purpose-driven APIs is what makes CDK L2s powerful. People who want the raw `string` format + other bare bones types use L1s.